player/command.c

```diff
@@ -6954,8 +6954,12 @@ void mp_option_change_callback(void *ctx, struct m_config_option *co, int flags,
         }
     }
 
-    if (opt_ptr == &opts->pause)
+    if (opt_ptr == &opts->pause) {
+        // If we are going to pause, always drain the ao on EOF if it exists.
+        if (mpctx->ao && mpctx->audio_status == STATUS_EOF && opts->pause)
+            ao_drain(mpctx->ao);
```

Comment:
hmm isn't there a better way to fit this into the audio state machine/code?
calling this from here smells like breaking abstraction

Reply:
Yes, quite an obvious place in retrospect. Thanks.